lib/spack/spack/compilers/aocc.py

```diff
@@ -106,7 +106,15 @@ def extract_version_from_output(cls, output):
         )
         if match:
             loc_ver = output.split('AOCC_')[1].split('-')[0]
-        return loc_ver
+            # Maintaining compatibility with previous release
+            # versioning format where dots are used instead
+            # of underscores.
+            # Only 3.1.0 is an expection, next release by default
+            # will have dots instead of underscores
+            if loc_ver == '3_1_0':
+                return '3.1.0'
+            else:
+                return loc_ver
```

Comment:
Can we use a better regex that covers both dots and underscores?

Reply:
Kindly note that underscore is a one off case at AOCC-3.1.0 release, in previous releases  we  had dots, going forward we will use only dots and not underscore.

Please note, we would like to keep it separate and not combine both underscore and dot.

Reply:
Yes, but you can use a regex that matches both `3.1.0` and `3_1_0` and reconstruct the version:
```diff
diff --git a/lib/spack/spack/compilers/aocc.py b/lib/spack/spack/compilers/aocc.py
index 6030743ad7..17b6546aa1 100644
--- a/lib/spack/spack/compilers/aocc.py
+++ b/lib/spack/spack/compilers/aocc.py
@@ -101,20 +101,11 @@ def extract_version_from_output(cls, output):
         loc_ver = 'unknown'
 
         match = re.search(
-            r'AMD clang version ([^ )]+)',
+            r'AOCC_(\d)[._](\d)[._](\d)',
             output
         )
         if match:
-            loc_ver = output.split('AOCC_')[1].split('-')[0]
-            # Maintaining compatibility with previous release
-            # versioning format where dots are used instead
-            # of underscores.
-            # Only 3.1.0 is an expection, next release by default
-            # will have dots instead of underscores
-            if loc_ver == '3_1_0':
-                return '3.1.0'
-            else:
-                return loc_ver
+            return '.'.join(match.groups())
 
     @classmethod
     def fc_version(cls, fortran_compiler):
```

Reply:
Incorporated suggested changes.